src/Servers/Kestrel/Core/src/CertificateLoader.cs

```diff
@@ -42,10 +42,15 @@ public static X509Certificate2 LoadFromStoreCert(string subject, string storeNam
                         .Where(DoesCertificateHaveAnAccessiblePrivateKey)
                         .OrderByDescending(certificate => certificate.NotAfter))
                     {
-                        foundCertificate = certificate;
+                        // Pick the first one as a fallback to substring default
+                        if (foundCertificate == null)
```

Comment:
You can replace this with `foundCertificate ??= certificate

Reply:
Thanks!